modules/loader-utils/src/lib/request-utils/request-scheduler.ts

```diff
@@ -132,14 +136,17 @@ export default class RequestScheduler {
 
   /** We check requests asynchronously, to prevent multiple updates */
   _issueNewRequests(): void {
-    if (!this.deferredUpdate) {
-      this.deferredUpdate = setTimeout(() => this._issueNewRequestsAsync(), 0);
+    if (this.deferredUpdate !== null) {
+      clearTimeout(this.deferredUpdate);
     }
+    this.deferredUpdate = setTimeout(() => this._issueNewRequestsAsync(), this.props.debounce);
```

Comment:
I am bit unsure about the update logic here, async stuff is easy to get wrong, tests are hard to write, and also the current code is battle tested. 

I am thinking could we just add the `debounceMS` on line 136 and not change anything more?

Reply:
If we only add `debounceMs` within the 'if' block, then this change will behave like a throttle rather than a debounce. Requests will be issued no more than once every X milliseconds, but the scheduler will not wait for new requests to stop coming.

That might also be reasonable, but it's a different behavior and would require different timeouts to get the user experience right, and we should probably rename the option to `throttleMs`... 

I suspect a debounce is somewhat better than a throttle for the use cases I'm after, but a throttle is certainly better than nothing, do you have a preference here?

Reply:
OK makes sense.